### PR TITLE
Documentation updates for environment setup on Linux and WSL

### PR DESCRIPTION
**Pull request by ivantarapov**
Updating documentation to include instructions on environment configuration for Linux and WSL deployments

Please follow the guidelines for PRs contained [here](docs/pull_requests.md). Checklist:

- [x] Your PR is small, and implements one change.
- [ N/A, doc-only] You have added unit tests for all functions that you modified.
- [ N/A, doc-only] You have run PyCharm's code cleanup tools on your Python files.
- [? ] Link the correct GitHub issue for tracking.
- [x] When merging your PR, replace the default merge message with a description of your PR,
and if needed a motivation why that change was required.

Note that I couldn't get WSL2 to work with CUDA - despite following the official instructions on NVidia's website, it never detects GPU. Not very surprising given that support is still in preview, but would be great if someone can check that.

### CODE REVIEW

File: docs/environment.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Setting up your environment
 
+## Prerequisites
+
+If you are planning to contribute to the solution, your OS environment will need [git](https://git-scm.com/) and [git lfs](https://git-lfs.github.com/) installed. Depending on the OS that you are running the installation instructions may vary. Please refer to respective documentation sections on the tools' websites for detailed instructions.
```
Comment (ant0nsc) [inline docs/environment.md:5]: You need git and git-lfs also if you just want to check the code out.

File: docs/environment.md [hunk 2]
```diff
@@ -37,11 +40,114 @@ You can skip this step if you have installed WSL as per the previous item.
...
 
-## Create a Conda environment
 Start the `conda` prompt for your platform. In that prompt, navigate to your repository root and run
 `conda env create --file environment.yml`
 
+## Using GPU locally
+
+It is possible to run the training process on a local machine. It will not be as performant as using a GPU cluster that Azure ML offers and you will not be able to take advantage of other Azure ML features such as comparing run results, creating snapshots for repeatable machine learning experiments or keeping history of experiment runs. At the same time it could be useful to experiment with code or troubleshoot things locally. 
+
+The SDK uses PyTorch to compose and run DNN computations. PyTorch can run on any CPU, but it is also capable of taking advantage of the underlying GPU via NVidia CUDA technology, which accelerates computations dramatically.
```
Comment (ant0nsc) [inline docs/environment.md:58]: I'd skip that, I assume everyone in the DNN space knows that.

File: docs/environment.md [hunk 3]
```diff
@@ -37,11 +40,114 @@ You can skip this step if you have installed WSL as per the previous item.
...
 
+## Using GPU locally
+
+It is possible to run the training process on a local machine. It will not be as performant as using a GPU cluster that Azure ML offers and you will not be able to take advantage of other Azure ML features such as comparing run results, creating snapshots for repeatable machine learning experiments or keeping history of experiment runs. At the same time it could be useful to experiment with code or troubleshoot things locally. 
+
+The SDK uses PyTorch to compose and run DNN computations. PyTorch can run on any CPU, but it is also capable of taking advantage of the underlying GPU via NVidia CUDA technology, which accelerates computations dramatically. 
+
+In order to enable PyTorch to use CUDA, you need to make sure that you have  
+a) Compatible graphics card with CUDA compute capability of at least 3.0 (at the moment of writing). You can check compatibility list here: https://developer.nvidia.com/cuda-gpus  
+b) Recent NVidia drivers installed
```
Comment (ant0nsc) [inline docs/environment.md:62]: a) and b) are non-standard lists in MD.

File: docs/environment.md [hunk 4]
```diff
@@ -37,11 +40,114 @@ You can skip this step if you have installed WSL as per the previous item.
...
+```
+$> wsl --list -v
+  NAME      STATE           VERSION
+* Ubuntu    Running         2
+```
+Note the "2" in Version column.
+
+At this point you need to install the compatible NVidia driver and CUDA toolkit. At the moment of writing the driver is still in preview, so issues are possible. Follow this guide for installation: https://docs.nvidia.com/cuda/wsl-user-guide/index.html
+
+You can also find a video walkthrough of WSL2+CUDA installation here: https://channel9.msdn.com/Shows/Tabs-vs-Spaces/GPU-Accelerated-Machine-Learning-with-WSL-2
```
Comment (ant0nsc) [inline docs/environment.md:92]: anything for WSL should go into WSL.md.

**Review by ant0nsc — APPROVED** [review]

File: docs/environment.md (hunk 1 above)
Comment (ivantarapov) [reply docs/environment.md]: Updated

File: docs/environment.md (hunk 2 above)
Comment (ivantarapov) [reply docs/environment.md]: Rephrased

File: docs/environment.md (hunk 3 above)
Comment (ivantarapov) [reply docs/environment.md]: Changed to ordered lists

File: docs/environment.md (hunk 4 above)
Comment (ivantarapov) [reply docs/environment.md]: Done

**Review by ant0nsc — APPROVED** [review]

**Review by javier-alvarez — APPROVED** [review]

**Pull request merged**